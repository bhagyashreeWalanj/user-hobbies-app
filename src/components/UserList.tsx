import * as React from 'react'
import styles from './../styles/UserList.module.scss'
import { default as classnames } from 'classnames'
import { IUser } from './../interfaces/user'

// Create the interface for UserList
interface UsersListProps {
  Users: IUser[]
  selectedUserId: string
  selectUser(id: string): void
}

const UserList = ({ Users, selectedUserId, selectUser }: UsersListProps) => {
  return (
    <div className={styles.listContainer}>
      <div>
        {Users.map((user) => {
          const isSelected = user.id === selectedUserId
          return (
            <div
              key={user.id}
              id={`username_${user.id}`}
              className={classnames({
                [styles.userListContainer]: true,
                [styles.selected]: isSelected,
              })}
              onClick={() => selectUser(user.id)}
            >
              <p className={styles.name}>{user.name}</p>
            </div>
          )
        })}
      </div>
    </div>
  )
}

export default UserList
