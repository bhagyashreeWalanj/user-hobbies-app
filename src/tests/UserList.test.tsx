import React from 'react'
import { mount, shallow, render } from 'enzyme'
import UserList from '../components/UserList'

describe('Testing User List Component', () => {
  const selectUser = jest.fn()
  const users = [
    {
      id: '1',
      name: 'John Hecker',
      hobbies: [
        {
          id: '2',
          hobby: 'Playing Cricket',
          passion: 'Low',
          year: '1990',
        },
      ],
    },
    {
      id: '2',
      name: 'John Hecker',
      hobbies: [
        {
          id: '2',
          hobby: 'Playing Cricket',
          passion: 'Low',
          year: '1990',
        },
      ],
    },
    {
      id: '3',
      name: 'John Hecker',
      hobbies: [
        {
          id: '2',
          hobby: 'Playing Cricket',
          passion: 'Low',
          year: '1990',
        },
      ],
    },
  ]

  const wrapper = shallow(
    <UserList Users={users} selectedUserId={''} selectUser={selectUser} />,
  )
  it('Renders user Hobbies Component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should render All users from database', () => {
    const userContainer = wrapper.find('.userListContainer')
    expect(userContainer.exists()).toBeTruthy()
    expect(userContainer.length).toBe(users.length)
  })

  it('should click on user name and call function for hobbies list', () => {
    wrapper.find('#username_1').simulate('click')
    expect(selectUser).toBeCalledTimes(1)
  })
})
