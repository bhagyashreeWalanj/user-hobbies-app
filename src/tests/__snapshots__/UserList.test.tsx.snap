// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing User List Component Renders user Hobbies Component 1`] = `
<div
  className="listContainer"
>
  <div>
    <div
      className="userListContainer"
      id="username_1"
      key="1"
      onClick={[Function]}
    >
      <p
        className="name"
      >
        John Hecker
      </p>
    </div>
    <div
      className="userListContainer"
      id="username_2"
      key="2"
      onClick={[Function]}
    >
      <p
        className="name"
      >
        John Hecker
      </p>
    </div>
    <div
      className="userListContainer"
      id="username_3"
      key="3"
      onClick={[Function]}
    >
      <p
        className="name"
      >
        John Hecker
      </p>
    </div>
  </div>
</div>
`;
